### PR TITLE
Fixed random stream reopen issue.

### PR DESCRIPTION
**Pull request by ppetrosh**

### CODE REVIEW

File: Twitter.agent.lib.nut [hunk 1]
```diff
@@ -120,25 +148,41 @@ class Twitter {
...
                 // Too many requests
                 // Try again with the _reconnectTimeout
-                imp.wakeup(_reconnectTimeout, function() { stream(searchTerms, onTweet, onError); }.bindenv(this));
-                _reconnectTimeout *= 2;
+                _reconnectTimer = imp.wakeup(_reconnectTimeout,
+                    function() {
+                        stream(searchTerms, onTweet, onError);
+                    }.bindenv(this)
+                );
+                _reconnectTimeout += TWITTER_DEFAULT_RECONNECT_TIMEOUT_SEC;
```
Comment (betzrhodes) [inline Twitter.agent.lib.nut]: This is a different calculation for backing off.  The original *=2 allows more time between reconnects than this += 60.

**Review by betzrhodes — COMMENTED** [review]
The bug fix looks good.  I've commented on the other change.

File: Twitter.agent.lib.nut (hunk 1 above)
Comment (ppetrosh) [reply Twitter.agent.lib.nut]: Right, and I believe it should be linear rather then exponential. But I may be wrong.

Comment (betzrhodes) [reply Twitter.agent.lib.nut]: This caused trouble in the past.

Comment (betzrhodes) [reply Twitter.agent.lib.nut]: If the account is getting too many tweets too fast the non linear had a better chance of recovering.

Comment (betzrhodes) [reply Twitter.agent.lib.nut]: This was implemented when the cowbell was getting hammered with retweets

Comment (ppetrosh) [reply Twitter.agent.lib.nut]: Oh I see. Thanks for catching this. Changed back to `*= 2;`

**Review by betzrhodes — APPROVED** [review]

**Pull request merged**